neurom/core/types.py

```diff
@@ -56,22 +59,27 @@ class SubtypeCollection(int):
     _BASE = 100
 
     def __new__(cls, *value):
-        """Create new objects."""
+        """Create an int representing a SubtypeCollection.
+
+        Args:
+            value (Union[int, Sequence[int], SubtypeCollection, NeuriteType, morphio.SectionType]):
+
+        """
         if len(value) == 1:
             # Avoid recursion error
             value = value[0]
 
-        if isinstance(value, (list, tuple)):
-            value = cls.from_list(value)
-        elif isinstance(value, Enum):
-            value = value.value
+        if isinstance(value, collections.abc.Sequence):
+            value = _ids_to_index([int(v) for v in value], cls._BASE)
+        else:
+            value = int(value)
 
         obj = super().__new__(cls, value)
 
-        if isinstance(value, SubtypeCollection):
-            obj._subtypes = list(value._subtypes)
-        else:
-            obj._subtypes = SubtypeCollection.to_list(obj)
+        obj._subtypes = tuple(
+            SectionType(int_type) for int_type in _index_to_ids(int(obj), cls._BASE)
```

Comment:
Why casting them to `SectionType`?

Reply:
Because these are the subtypes. Integers are not subtypes, just a number of convention to represent them. The integers may (unlikely) but the types won't.

Reply:
Ok